src/gpu/GpuSieve.hpp

```diff
@@ -0,0 +1,27 @@
+#ifndef GPU_SIEVE_HPP
+#define GPU_SIEVE_HPP
+
+#include "../interfaces/ISieve.hpp"
+#include <vector>
+#include <cstdint>
+
+class GpuSieve : public ISieve {
+public:
+    GpuSieve();
+    ~GpuSieve() override;
+
+    void setMaxLimit(unsigned int maxLimit) override;
+    void computePrimes() override;
+    const std::vector<unsigned int>& getPrimes() const override;
+    void collectPrimes() override;
+
+private:
+    unsigned int maxLimit;
```

Comment:
add underscores suffixes for any private member, in this case: unsigned int maxLimit_;